org.jacoco.core/src/org/jacoco/core/internal/runtime/ClassInjectionRuntime.java

```diff
@@ -0,0 +1,297 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.runtime;
+
+import java.lang.instrument.Instrumentation;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.jacoco.core.runtime.AbstractRuntime;
+import org.jacoco.core.runtime.IRuntime;
+import org.jacoco.core.runtime.RuntimeData;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+/**
+ * {@link IRuntime} that uses introduced in Java 9
```

Comment:
Would prefer to describe what the class does: "`IRuntime` which injects a new class into the `java.base` module for JaCoCo runtime access using Java 9 APIs."

Reply:
> for JaCoCo runtime access

is IMO redundant, because this is what all `IRuntime` implementations do.
Kept the rest.

Reply:
Ok.